File: Algorithms/Dynamic-Programming/Gauss_Easter_Date/Gauss_Easter_Date.py
import math

def Easter(Year):
    # Gauss Easter Algorithm
    a = Year % 19
    b = Year % 4
    c = Year % 7
    k = math.floor(Year / 100)
    p = math.floor((13 + 8 * k) / 25)
    q = k/4
    m = (15 - p + k - q) % 30
    n = (4 + k - q) % 7
    d = (19 * a + m) % 30
    e = (2 * b + 4 * c + 6 * d + n) % 7
    days = (22 + d + e)
  
    # d is 29
    if ((d == 29) and (e == 6)):
        print(Year, "- 04 -19")
        return
    # d is 28
    elif ((d == 28) and (e == 6)):
        print(Year, "- 04 -18")
        return
    else:
        # If days > 31, April
        if (days > 31):
            print(Year, "- 04 -", int(days - 31))
            return
        else:
            # Otherwise, March
            print(Year, "- 03 -", days)
            return




Year = 2012
Easter(Year)

File: Algorithms/Dynamic-Programming/Gauss_Easter_Date/Gauss_Easter_Date.rb
def gauss_easter(y)
    a = y % 19
    b = y % 4
    c = y % 7
    
    p = (y / 100).floor
    q = ((13 + 8 * p) / 25).floor
    m = (15 - q + p - (p / 4)) % 30
    
    n = (4 + p - (p / 4)) % 7
    
    d = (19 * a + m) % 30
    
    e = (2 * b + 4 * c + 6 * d + n) % 7
    
    day = (22 + d + e)
      
    if (d == 29) && (e == 6)
        puts("The Easter date for your entered year is: #{y}-04-19")
    elsif (d == 28) && (e == 6)
        return puts("The Easter date for your entered year is: #{y}-04-18")
    else
        if day > 31
            return puts("The Easter date for your entered year is: #{y}-04-#{day-31}")
        else
            return puts("The Easter date for your entered year is: #{y}-03-#{day}")
        end
    end    
end

print("Enter the year you want to calculate the Easter date: ")
y = Integer(gets.chomp())
gauss_easter(y)